java/client/src/main/java/glide/api/BaseClient.java

```
@@ -383,7 +380,12 @@ public PubSubMessage tryGetPubSubMessage() {
                     "The operation will never complete since messages will be passed to the configured"
                             + " callback.");
         }
-        return messageQueue.poll();
+        try {
+            return messageQueue.pop().get();
+        } catch (Exception unreachable) {
+            // should be never happen
+            return null;
```

If we need to get rid of this, need to add `popSync` and `popAsync`, which makes state machine of PSMQ more complex.

---

we cannot hide exceptions from the user code

---

No exceptions anymore 🎉

---

@Yury-Fridlyand 
I dont understand, the code looks the same,  why is it still catches are returns null. It should not catch or catch and re throw

---

@ikolomi , I don't see this in the current code (in the Files changed view). It just calls popSync() now with no try/catch.